samples/EventHub/allowed-event-hub-firewall-ip/azurepolicy.json

```diff
@@ -0,0 +1,39 @@
+{
+  "type": "Microsoft.Authorization/policyDefinitions",
+  "name": "allowed-event-hub-firewall",
+  "properties": {
+    "displayName": "Allow IP for event hub firewall",
+    "policyType": "Custom", 
+    "description": "This policy allows only restricted IPs for Event hub firewall",
+    "mode":"All",
+  "parameters": {
+    "allowedIps": {
+      "type": "Array",
+      "metadata": {
+        "displayName": "Allowed IPs for event hub firewall",
+        "description": "List of IPs allowed for event hub firewall"
+      },
+      "defaultValue": []
+    }
+  },
+  "policyRule": {
+    "if": {
+      "allOf": [
+        {
+          "field": "type",
+          "equals": "Microsoft.EventHub/namespaces/networkRuleSets"
+        },
```

Comment:
what if the defaultAction is set to "Allow"?  Then every IP is allowed.  Also, don't you need to take the ipRules.action into account as well?

Reply:
Added for below for considering only Deny conditions in policyrule
{
"field": "Microsoft.EventHub/namespaces/networkRuleSets/default.defaultAction",
"equals": "Deny
 }

Reply:
Thats not really what you want though right?  As written this policy will ONLY deny if defaultAction=deny and ipRules contains an IP not in allowedIPs.  The title and the description imply that if this policy was assigned ONLY a subset (or all) of the allowedIPs should have access to an event hub.  The policy should deny if defaultAction=allow

---
In reply to: [310361757](https://github.com/Azure/azure-policy/pull/448#discussion_r310361757) [](ancestors = 310361757)